exporter/kafkaexporter/factory_test.go

```diff
@@ -155,10 +217,14 @@ func TestCreateLogExporter(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
 
-			f := NewFactory()
+			core, observed := observer.New(zapcore.DebugLevel)
+			exporterSettings := componenttest.NewNopExporterCreateSettings()
+			exporterSettings.Logger = zap.New(core)
```

Comment:
I am still not a fan of using log statements to validate a setting, what I suggest you do is use the mock marshaler and call the exporter.

Reply:
Then the choices are to either add a config setting that tells the producer to noop or abstract the producer such that we can inject a noop impl for the test ([my original PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13433)).  Do you have a preference?

Reply:
Right, so it took me a bit to figure out why I wasn't certain on first PR. 

I would rather a safe default, instead nil implying default.

So a Noop type following what I had done here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awskinesisexporter/exporter.go#LL56 would be perfect if you have the time.

Reply:
It looks like my PR got merged in thinking yours got stale.

Reply:
Removed log-based validation of marshaler